### PR TITLE
Blog post "newbie in rdo", differences between rdo trunk and rdo stable

### PR DESCRIPTION
**Pull request by amoralej**
I'm preparing a set of blog posts that help new "RDOers" to get an idea of the big picture, what is produced, a high level view of the process, etc... This is the first post. 

Note that I've write it thinking in the new repos for that will be served from buildlogs and hashed repos and reports from trunk.rdo, so it shouldn't be published until this is ready.

### CODE REVIEW

File: source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+- **rdo-trunk-&lt;branch>:** provides the latest snapshot of packages successfully built (i.e. rdo-trunk-mitaka).
+- **rdo-trunk-&lt;branch>-tested:** provides the latest snapshot of packages built and with CI jobs successfully passed (i.e. rdo-trunk-master-tested).
+- **rdo-trunk-master-tripleo:** provides the latest snapshot of packages tested by specific TripleO CI jobs on master branch.
+
+Currently (apr '16'), &lt;branch> can be master, mitaka, liberty or kilo.
+
+As operators, these repos may not be the best option to run a stable OpenStack cloud but it can be a good choice if you are interested in deploying an environment to test latest changes in OpenStack code. If you are a OpenStack developer it may be a good option to build environments with latest vanilla upstream code.
```
Comment (apevec) [inline source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md]: RDO trunk for stable branches could be an option for operators, when quick hotfix merged to upstream stable branch is needed. Once we have semver fix in DLRN and upgrades back/forth are tested, rdo-trunk-{mitaka|liberty}-tested can be advertised targeting that use-case.

File: source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+Finally, if you are a packager, you probably want to know how the build process went for a specific commit or even to perform an installation based on it. RDO provides both build reports and RPM repos based on each commit for both CentOS and Fedora from https://trunk.rdoproject.org.
+
+[Fred](http://blogs.rdoproject.org/author/fred) wrote an excelent [post](http://blogs.rdoproject.org/7834/delorean-openstack-packages-from-the-future) describing how DLRN (f.k.a. delorean) build packages for RDO trunk repos.
+
+### RDO Stable repos
+Provide a more stable set of packages throguh CentOS CloudSIG repos based on [CentOS Community Build System](https://wiki.centos.org/HowTos/CommunityBuildSystem). These repos differs from RDO Trunk in several ways:
```
Comment (apevec) [inline source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md]: throguh -> through

File: source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+- Packages are only created after GA of a upstream release (only stable branches are used, not master).
```
Comment (apevec) [inline source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md]: created -> published (we do have later milestones and RCs builds)

File: source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+- New packages are created only when a new release is published (release tag created) on upstream repositories. Rebuild for those packages is not automatically triggered so it may take some time to deliver them.
```
Comment (apevec) [inline source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md]: new release -> new point release

Comment (apevec) [inline source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md]: not automatically -> plan is to automate as much as possible in rpmfactory

File: source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+- In addition to the vanilla upstream code, some patches may be backported to fix security issues or critical bugs not backported upstream. Note that an upstream-first policy is applied so these patches will be applied only after merged in upstream master.
```
Comment (apevec) [inline source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md]: In addition to critical security and show-stopper issues, also allowed are source code patches to help packaging, prime example is oslo.config patch[1] to enabled distro defaults from dist.conf files

[1] https://github.com/openstack-packages/python-oslo-config/blob/rpm-master/0001-add-usr-share-project-dist.conf-to-the-default-confi.patch

File: source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+- Packages are created through a more controlled and reproducible workflow.
```
Comment (apevec) [inline source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md]: s/more// - you can shortly explain how Koji buildsystem ensures reproducibility: no network access from buildroot, requiring all BuildRequires to be packaged and by recording both spec (NVR must be unique) and buildroot repo version (NVRs of all RPMs in buildroot is recorded).

File: source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
+---
+title: "Newbie in RDO: one size doesn't fit all"
+author: amoralej
+tags: openstack, rdo, repos
+date: 2016-04-08 19:50:56 CEST
+---
+
+As a new contributor to RDO project, one of my first tasks was to understand what is actually being produced by the project. According to [RDO faq](https://www.rdoproject.org/rdo/faq/), the main goal is to maintain a freely-available, community-supported distribution of OpenStack based on RPM packages able to run Red Hat Enterprise Linux, CentOS, Fedora, and their derivatives. But RDO community is diverse and different use patterns have been found over the time with different requirements:
```
Comment (javierpena) [inline source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md]: to the RDO project
<nip>
But the RDO community

File: source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+- Organizations running production OpenStack clouds with community support require a stable distribution.
+- Organizations running production environments with community or enterprise-grade distributions require sometimes an up-to-date distribution to test recent functionalities and plan the future of their cloud deployments.
+- Developers require up-to-date distribution to test their changes and run their CI jobs.
+- Packagers need to have full control over the packaging process.
+
+Provide a distribution suitable for all these scenarios is not an easy task and RDO community ended up with two different distribution set of repos targetting different purposes:
```
Comment (javierpena) [inline source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md]: Providing a suitable distribution... and the RDO community...

File: source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+### RDO Trunk repos
+Provide RPM package repositories for OpenStack projects as close as possible to the vanilla upstream code in order to test any change and catch any bug or impact on packaging as soon as possible. These packages are built when a commit is merged in upstream master or any of the supported stable branches. Once built is done, a set of CI jobs is passed to validate installation and basic functionalities. As part of RDO Trunk we provide different repos for CentOS 7 under http://buildlogs.centos.org/centos/7/cloud/x86_64/ :
```
Comment (javierpena) [inline source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md]: Once a build is done

File: source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md (hunk 2 above)
Comment (javierpena) [inline source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md]: There repos differ from RDO Trunk...

File: source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+### RDO Stable repos
+Provide a more stable set of packages throguh CentOS CloudSIG repos based on [CentOS Community Build System](https://wiki.centos.org/HowTos/CommunityBuildSystem). These repos differs from RDO Trunk in several ways:
+
+- Packages are only created after GA of a upstream release (only stable branches are used, not master).
+- New packages are created only when a new release is published (release tag created) on upstream repositories. Rebuild for those packages is not automatically triggered so it may take some time to deliver them.
+- In addition to the vanilla upstream code, some patches may be backported to fix security issues or critical bugs not backported upstream. Note that an upstream-first policy is applied so these patches will be applied only after merged in upstream master.
+- Packages are created through a more controlled and reproducible workflow.
+- The criteria to pubishing any new release includes bot thechnical (a set of CI jobs must pass as aggreed in [RDO community meeting](https://meetbot.fedoraproject.org/rdo/2016-04-06/rdo_meeting_%282016-04-06%29.2016-04-06-15.00.html)) and non-technical requirements (documentation, etc...).
```
Comment (javierpena) [inline source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md]: ...includes both technical...

File: source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md (hunk 1 above)
Comment (amoralej) [inline source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md]: When we expect to have the new versioning in place? is it worthy to add it in the post?

**Comment by amoralej** [comment]
Just included all your improvements and fixes.

File: source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+### RDO Trunk repos
+Provide RPM package repositories for OpenStack projects as close as possible to the vanilla upstream code in order to test any change and catch any bug or impact on packaging as soon as possible. These packages are built when a commit is merged in upstream master or any of the supported stable branches. Once a build is done, a set of CI jobs is passed to validate installation and basic functionalities. As part of RDO Trunk we provide different repos for CentOS 7 under http://buildlogs.centos.org/centos/7/cloud/x86_64/ :
+
+- **rdo-trunk-&lt;branch>:** provides the latest snapshot of packages successfully built (i.e. rdo-trunk-mitaka).
+- **rdo-trunk-&lt;branch>-tested:** provides the latest snapshot of packages built and with CI jobs successfully passed (i.e. rdo-trunk-master-tested).
+- **rdo-trunk-master-tripleo:** provides the latest snapshot of packages tested by specific TripleO CI jobs on master branch.
+
+Currently (apr '16'), &lt;branch> can be master, mitaka, liberty or kilo.
```
Comment (apevec) [inline source/blog/2016-04-08-new-in-rdo-repos-one-size-doesn-t-fit-all.html.md]: @amoralej update this line before publishing to include only master, mitaka and liberty.
Kilo is EOL now.

**Pull request merged**